app/src/org/commcare/activities/QueryRequestUiController.kt

```diff
@@ -176,18 +176,27 @@ class QueryRequestUiController(
             override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
             override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
             override fun afterTextChanged(s: Editable) {
-                remoteQuerySessionManager.answerUserPrompt(queryPrompt.key, s.toString())
+                answerUserPrompt(queryPrompt.key, s.toString())
                 updateAnswerAndRefresh(queryPrompt, s.toString())
             }
         })
         return promptEditText
     }
 
+    private fun answerUserPrompt(key: String, value: String) {
+        // todo mobile don't support blank search yet
+        if (value == "") {
```

Comment:
Any particular reason for not comparing this with `equals`?

Reply:
not really, can change.

Reply:
same. the issue there is depending on how `value` is passed that comparison can fail, like with `new String("")`, JVM won't use any interned literals.